Docs 6761 on premises v4.2 (#1044)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/deploy-on-premises.adoc

The Mule Maven plugin also supports deploying domains when using the standalone deployment strategy, or the Runtime Manager agent deployment strategy.

> There are also additional operations that enable you to deploy applications in parallel and update applications at runtime.

Reviewer: ```suggestion
There are also additional operations that enable you to deploy applications in parallel and update applications at runtime.
```
```suggestion
Additional operations enable you to deploy applications in parallel and update applications at runtime.
```

mvn clean package deploy -DmuleDeploy
----

== Deploy Applications in Parallel

You can deploy applications in parallel to on-premises Mule instances. Deploying applications in parallel reduces the startup time when you deploy a large number of apps. To prevent deployment failure, ensure that applications are not dependent on each other because a particular start order cannot be guaranteed.

To enable parallel deployment:

. Package or export your Mule applications to the `/apps` directory of your Mule runtime engine instance.
. Start Mule using the `-M-Dmule.deployment.parallel` option:
+
```
mule -M-Dmule.deployment.parallel
```

After performing these steps, all applications in the `/apps` directory start in parallel.

== Update Applications at Runtime

> Updating a Mule application at runtime can be a complex change involving class modifications, endpoint modifications (for example, changing ports, etc), as well as reconfiguring flows. As a result, any application update does a graceful app shutdown under the hood and reconfigures itself. In practice, this is pretty transparent to the user and happens within seconds.

Reviewer: ```suggestion
Updating a Mule application at runtime can be a complex change involving class modifications, endpoint modifications (for example, changing ports), and reconfiguring flows. To keep this process from disrupting users, an application update first performs a graceful app shutdown and then reconfigures itself transparently and within seconds.
```
Or similar, if this alters intended meaning. The troublesome phrase was "as a result," because I think that we want to refer to the transparency as the result and the shutdown/reconfigure as the way to address the complexity?

> There are several ways you can update an application:

Reviewer: "several" = 2, which is more like "a couple of ways" or just "two ways"....

> * By dropping the modifications over an existing exploded app folder and touching the master configuration file (`mule-config.xml` located in the app's root directory by default).

Reviewer: I am not familiar (yet) with "dropping," "exploded folder," or "touching a file," so I don't understand this enough to suggest a possible rewording of those terms. Check with a writer peer?

Reviewer: Is "dropping" the same as "adding"?

> * By dropping a new `jar` version of the app into the `$MULE_HOME/apps` directory. Mule detects the `jar` as an existing application and ensures the updated by a clean redeployment of the app. +

Reviewer: Per our recommended writing style, break list items and steps after one sentence:
```suggestion
* By dropping a new `jar` version of the app into the `$MULE_HOME/apps` directory. 
Mule detects the `jar` as an existing application and ensures the updated by a clean redeployment of the app. +
```
Also, a couple of questions about the meaning:
"a new `jar` version of the app: is this a JAR file replacement for the app? The app is in a JAR file?
Does Mule detect this as an UPDATE and overwrite the existing app file?

> Note that Mule discards any modifications to the old application folder. The new app folder is a clean exploded application from a `jar`.

Reviewer: Again, not certain about "clean exploded application."
Are we saying that adding the new JAR file (folder?) replaces the existing (old) folder, including any changes to that old folder?
Again, suggest having a peer writer take a look at the wording, since it is so new to me.

> As you see, both integrate pretty well with existing build tools, the preference for one over another really depends on established development practices only.

Reviewer: ```suggestion
Either method integrates well with existing build tools.
```
